interview-questions.md

```diff
@@ -4,38 +4,50 @@ Answer the following questions. First, without external resources. Challenge you
 
+    Researched answer: There are single-line blocks and multi-line blocks. If you want to use a multi-line block, it is recommended that you use the do/end notation as that keeps it cleaner. There are many applications that can be done with blocks such as using the yield keyword and converting them to Procs or Lambdas to save time and coding space. 
 
+    source: https://mixandgo.com/learn/mastering-ruby-blocks-in-less-than-5-minutes
 
 3. Ruby has an implicit return. What does that mean?
 
-  Your answer:
-
-  Researched answer:
+    Your answer: In Ruby, you don't need to write the word "return" like we do in JavaScript. It is implied that after the do or {, the code written in the block will be returned. 
+    If there are multiple cases of code, such as if/elsif statements, the last line, usually the else statement, will not need a return since Ruby automatically returns the final code block statement. 
 
+    Researched answer: Any statement in Ruby returns the value of the last evaluated expression.
 
+    source: https://www.freecodecamp.org/news/idiomatic-ruby-writing-beautiful-code-6845c830c664/
 
 4. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+    Your answer: Object Oriented Programming (OOP) is programming where everything (or almost everything) is considered an object. The interactions between the data is interactions between the classes and objects and the methods within them. 
+
+    Researched answer: The four principles of object-oriented programming are encapsulation, abstraction, inheritance, and polymorphism.
+
+    Encapsulation is achieved when each object keeps its state private, inside a class. Other objects don’t have direct access to this state. Instead, they can only call a list of public functions — called methods.
+
+    Applying abstraction means that each object should only expose a high-level mechanism for using it. This mechanism should hide internal implementation details. It should only reveal operations relevant for the other objects. Think — a coffee machine. It does a lot of stuff and makes quirky noises under the hood. But all you have to do is put in coffee and press a button. 
 
-  Researched answer:
+    Inheritance: it means that you create a (child) class by deriving from another (parent) class. This way, we form a hierarchy.The child class reuses all fields and methods of the parent class (common part) and can implement its own (unique part). 
 
+    Polymorphism gives a way to use a class exactly like its parent so there’s no confusion with mixing types. But each child class keeps its own methods as they are.
 
+    source: https://www.freecodecamp.org/news/object-oriented-programming-concepts-21bb035f7260/
+    **really enjoyed reading this learning more about the concepts that make up OOP.**
```

Comment:
great! instructors will look at this and try and implement it in class if it was a good resource :)